plan_delegator/include/plan_delegator.hpp

```diff
@@ -33,18 +43,105 @@
 
 namespace plan_delegator
 {
-    /**
-     * Mock plan delegator shell for initial testing.
-     */
     class PlanDelegator
     {
         public:
+
+            // constants definition
+            static const constexpr double MILLISECOND_TO_SECOND = 0.001;
+
+            PlanDelegator();
+
             /**
-             * \brief Initialize the plan delegator and run it
-             * \return Standard unix program exit code
+             * \brief Initialize the plan delegator
              */
-            int run();
+            void init();
+
+            /**
+             * \brief Run the main thread of plan delegator
```

Comment:
If there's no multithreading here (i.e. this method just runs in the main process) then I'd change the way this is phrased to reflect that, maybe `Run the main loop of the plan delegator` or something like that.

Reply:
Sure. Updated.